src/rgw/rgw_common.cc

```diff
@@ -513,7 +513,7 @@ bool parse_rfc2616(const char *s, struct tm *t)
 bool parse_iso8601(const char *s, struct tm *t, uint32_t *pns, bool extended_format)
 {
   memset(t, 0, sizeof(*t));
-  const char *p;
+  const char *p = nullptr;
```

Comment:
@joscollin: to be honest, I don't see a substantial difference before and after the change. The *from-C-string* constructor of `boost::string_view` IMO never saw the uninitialized `p`. Let me quote from `man 3 strptime`:
> **RETURN VALUE**
The  return  value  of  the  function  is a pointer to the first character not processed in this function call.  In case the input string contains more characters than required by the format string, the return value points right after the last consumed input character.  In case the whole input string is  consumed,  the  return  value points to the null byte at the end of the string.  If `strptime()` fails to match all of the format string and therefore an error occurred, the function returns `NULL`.

So, the result appears as well-defined. Even on the error path the `strptime` should return `nullptr` which would be caught by the check:
```cpp
   if (!p) {
     dout(0) << "parse_iso8601 failed" << dendl;
     return false;
}
```